plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/DoubleDecoder.java

```diff
@@ -29,6 +29,10 @@ public void decode(Supplier<Object> getter, BlockBuilder output)
         if (value == null) {
             output.appendNull();
         }
+        else if (value instanceof String) {
+            // Pinot returns NEGATIVE_INFINITY, POSITIVE_INFINITY as a String
+            DOUBLE.writeDouble(output, Double.valueOf(value.toString()));
```

Comment:
Also if `value` is a String is the `value.toString()` really needed? A cast should do.